index.d.ts

```diff
@@ -22,17 +24,34 @@ export type WithDataProps =
   }
   & WithData
 
+export type Matching<InjectedProps, DecorationTargetProps> = {
+  [P in keyof DecorationTargetProps]: P extends keyof InjectedProps
+      ? InjectedProps[P] extends DecorationTargetProps[P]
+          ? DecorationTargetProps[P]
+          : InjectedProps[P]
+      : DecorationTargetProps[P];
+};
 
-type MapRecordsToPropsFn<TWrappedProps> = (props: TWrappedProps) => RecordsToProps;
+export type GetProps<C> = C extends ComponentType<infer P> ? P : never;
 
-export type MapRecordsToProps<TWrappedProps> =
-  | RecordsToProps
-  | MapRecordsToPropsFn<TWrappedProps>
+type MapRecordsToPropsFn<TRecordProps = {}, TOwnProps = {}> = (props: TOwnProps) => RecordsToProps<keyof TRecordProps>;
+
+export type MapRecordsToProps<TRecordProps = {}, TOwnProps = {}> =
+  | RecordsToProps<keyof TRecordProps>
+  | MapRecordsToPropsFn<TRecordProps, TOwnProps>
 
 export class DataProvider extends React.Component<DataProviderProps> {}
 
+export type ConnectedComponentClass<C, P> = ComponentClass<JSX.LibraryManagedAttributes<C, P>> & {
+  WrappedComponent: C;
+};
+
+export interface InferableComponentEnhancerWithProps<TInjectedProps, TNeedsProps> {
```

Comment:
Can't really figure it out by myself... ;)
What does this interface do?

Reply:
I copy pasted from react-redux. It infers the props of the component and the props that it is going to inject and removes the props that it injects from the props interface. Next to that it makes sure that the types of things like displayName and propTypes don’t get lost.